feat: allow using variables in fields of webhook payload template

valekjo opened the pull request:
This PR
 - makes parsing slightly more effective (one pass only)
 - adds support for "interpolation" - using `{{var:resource.defaultDatasetId}}` syntax in strings - ie. adding the possibility to pass it to the input.
 - It's intentionally slightly different mechanism than the parsing, as it does a different thing.
 
 Theoretically it's breaking, I have an idea how to deal with it, just not sure if it's necessary.

fnesveda commented on packages/utilities/src/webhook_payload_template.ts:
```diff
@@ -125,31 +133,74 @@ export class WebhookPayloadTemplate {
+
+    private _interpolateString(value: string): string {
+        // If the string matches exactly, we return the variable value including the type
+        if (value.match(/^\{\{var:([a-zA-Z0-9.]*)\}\}$/)) {
```
This would make `{{var:}}` valid too, you should have `+` instead of `*` in the regex

valekjo replied:
fixed

fnesveda reviewed (changes requested):
Looks good, just a few comments mostly about the tests. Plus we talked live about maybe dropping the `var:` prefix.

valekjo commented:
> Looks good, just a few comments mostly about the tests. Plus we talked live about maybe dropping the `var:` prefix.

@fnesveda I've updated the tests, found one bug during the process (how replacement works). And figured out one more reason why I added the `var:` prefix - so it doesn't break the original behaviour. So instead of `var:` there is new `interpolateStrings` option, which just turns on/off the parsing in strings.

fnesveda commented on packages/utilities/src/webhook_payload_template.ts:
```diff
@@ -125,31 +140,75 @@ export class WebhookPayloadTemplate {
+    private _interpolateString(value: string): string {
+        // If the string matches exactly, we return the variable value including the type
+        if (value.match(/^\{\{([a-zA-Z0-9.]+)\}\}$/)) {
+            // This just strpis the {{ and }}
```
```suggestion
            // This just strips the {{ and }}
```

fnesveda reviewed (approved):
Looks good, one tiny typo
